### DISCUSSION TITLE
How can I exclude a local library package from being part of the PEX dependencies cache?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18276] [Q&A] [answered]
I'm having trouble deploying dagster project using the `dagster-cloud` cli. The deployment hasn't changed. I suspect it's related to importing a local package as a library in the setup.py:
``` install_requires=[
        # Local dependencies
        "dagster-lib @ file://"
        + os.path.join(os.path.dirname(__file__), "../libs/dagster_lib"),

```

I get:

```
dagster._core.errors.DagsterImportError: Encountered ImportError: cannot import name 'SqlIOManager' from 'dagster_lib.resources.io_managers' (/venvs/1a7e8cb1c6a5/lib/python3.10/site-packages/dagster_lib/resources/io_managers.py)
while importing module data_pipelines. Local modules were resolved using the working directory `/venvs/4c52386641a1/lib/python3.10/site-packages/working_directory/root`. If another working directory should be used, please explicitly specify the appropriate path using the `-d` or `--working-directory` for CLI based targets or the `working_directory` configuration option for workspace targets. 
```

 In the deployment logs I notice:
```
Reusing cached dependencies: deps-dabf9cc84304f18af0dce6c9b35b3c8730259878.pex
```

Removing

`--deps-cache-from`

in the cli args fixed this.
So now my question is: how can I exclude my local library package from being part of the cache?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C02LJ7G0LAZ/p1701092904270159?thread_ts=1701092904.270159&cid=C02LJ7G0LAZ).*

**Comment by shalabhc** [comment, marked as answer]
Unfortunately this is not currently possible. All dependencies outside the project directory are included in the cache.

If you do not make frequent modifications to the local dependency (`dagster_lib`) then you can continue to use `--deps-cache-from` for faster deployments. Whenever you modify the local dependency, you can run the command once without `--deps-cache-from` to force a dependency rebuild.

Alternatively you can copy the local dependency into your project folder before the build, and remove it from the `setup.py`. This will deploy that module with the project and it not be part of the cache.

**Comment by bmarcj** [comment]
Thanks @shalabhc . 

Not ideal, but both of those workarounds as clunky but possible (probably the first, as the library rarely changes). Potentially there is a room in ci/cd to identify a library change and disable the cache when that happens, but I've yet to work that out.
